builtin/credential/ldap/backend.go

```diff
@@ -184,9 +190,19 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 	// Merge local and LDAP groups
 	allGroups = append(allGroups, ldapGroups...)
 
+	canonicalGroups := make([]string, len(allGroups))
```

Comment:
Can we avoid this allocation by directly editing `allGroups`?

Reply:
Actually whether the right thing to do is modify allGroups or not depends on whether we want Identity group aliases to use canonicalized lowercase groups or groups as passed back from the LDAP server.

Reply:
I think it makes sense to reflect what was returned from the server.